src/curl.c

```diff
@@ -363,6 +364,62 @@ int swupd_curl_get_file(const char *url, char *filename, struct file *file,
 	return err;
 }
 
+void swupd_curl_test_resume(void)
+{
+#define RESUME_BYTE_RANGE 2
+
+	CURLcode curl_ret;
+	char *version_string = NULL;
+	char *url = NULL;
+
+	if (!curl) {
+		abort();
+	}
```

Comment:
I agree in both, we need to do the change in all code and we need to do it
in a separate change, I can follow up and do it in a separate patch.

On Thu, Apr 28, 2016 at 1:49 PM, Patrick McCarty <notifications@github.com>
wrote:

> In src/curl.c
> <https://github.com/clearlinux/swupd-client/pull/59#discussion_r61482062>:
>
> > @@ -363,6 +364,62 @@ int swupd_curl_get_file(const char *url, char *filename, struct file *file,
> >  	return err;
> >  }
> >
> > +void swupd_curl_test_resume(void)
> > +{
> > +#define RESUME_BYTE_RANGE 2
> > +
> > +	CURLcode curl_ret;
> > +	char *version_string = NULL;
> > +	char *url = NULL;
> > +
> > +	if (!curl) {
> > +		abort();
> > +	}
>
> I would prefer to do "assert(curl)" here instead, since it indicates a
> programming error if curl is not initialized at this point in the code.
>
> But, the code is not consistent in this regard, so maybe we should make
> this type of change throughout the code base at once in a separate commit.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/clearlinux/swupd-client/pull/59/files/eecdce9de214c06e806037ec1cf24e17f8b8e7e4#r61482062>
>